patch-core.js

```diff
@@ -11,12 +11,35 @@ const https = require('https');
 const patchMarker = "__agent_base_https_request_patched__";
 if (!https.request[patchMarker]) {
   https.request = (function(request) {
-    return function(_options, cb) {
-      let options;
+    return function(_url, _options, cb) {
+      // Support for both
+      // https.request(options[, callback])
+      // https.request(url[, options][, callback])
+      if (typeof _url !== 'string' && !cb) {
+        cb = _options;
+        _options = _url;
+        _url = undefined;
+      }
+      if (!_url && _options) {
+        _url = url.format(_options);
```

Comment:
Why do we do this? Which case are we handling here?

Reply:
https.request has two valid behaviors:

```
https.request(options[, callback])
https.request(url[, options][, callback])
```

For that reason, if _url isn't a string and there's no callback, between line 19 and 21 we're assuming the scenario with only two parameters was used, so we're re-assigning the parameters to make sense with their correspondent name. Right at the end of that, I'm assigning `undefined` to `_url`, trying to simplify what we're trying to accomplish in that `if` block.

After that, I'm only checking that if we don't have a `_url` and we do have `_options`, we can now try to transform the `_options` to have a string `_url`. We do this by using `url.format`, which returns a string and works for both legacy URLs and WHATWG URLs. This second `if` block also helps if the user decided to pass a falsy `_url`, since any form of a falsy `_url` will break this anyway.

Feedback appreciated! 💚